packages/core-app-api/src/lib/AuthConnector/DefaultAuthConnector.ts

```diff
@@ -93,7 +106,7 @@ export class DefaultAuthConnector<AuthSession>
   }
 
   async createSession(options: CreateSessionOptions): Promise<AuthSession> {
-    if (options.instantPopup) {
+    if (options.instantPopup && this.authFlow === 'popup') {
```

Comment:
The redirect will be executed when authFlow is set to redirect since the auth requester will perform the redirect when invoked. Please let me know if you think the logic still should be updated here.